### PR TITLE
[Compute] Upgrade disk API version to 2020-06-30 and some new features

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

Resolve https://github.com/Azure/azure-cli/issues/14899

[Compute] Upgrade disk API version to 2020-06-30

[Compute] `disk create`: add `--logical-sector-size`, `--tier`
Logical sector size in bytes for Ultra disks. Supported values are 512 ad 4096. 4096 is the default.

[Compute] `disk update`: Support `--disk-iops-read-only`, `--disk-mbps-read-only`, `--max-shares`

[Compute] new command `disk-encryption-set list-associated-resources`

**Testing Guide**  

**History Notes**  

[Compute] Upgrade disk API version to 2020-06-30
[Compute] `disk create`: add `--logical-sector-size`, `--tier`
[Compute] `disk update`: Support `--disk-iops-read-only`, `--disk-mbps-read-only`, `--max-shares`
[Compute] new command `disk-encryption-set list-associated-resources`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Review by fengzhou-msft — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py:3985]: Will this always be skipped in CI?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py:3985]: It is temporary skip. I can't run it lively. After the PR is merged, I will open issues for them to fix and rerun them. There are 4 test cases that I skip.

**Comment by qwordy** [comment]
Create an issue to track failed tests.
https://github.com/Azure/azure-cli/issues/15097

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py
Comment (andyzhangx) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:137]: is `2020-06-1` api-version available on all regions including sovereign cloud, e.g. Azure China?
@VybavaRamadoss

Comment (andyzhangx) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py:137]: /hold
hold on a moment for confirm, thanks.

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py
Comment (arrownj) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py:3985]: Are you sure all the skipped cases are all only environment problem ?

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_custom.py
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_custom.py:668]: Why need to update `acs` and `cosmosdb` cases in [Compute] PR ?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_custom.py:668]: Disks are used everywhere.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_custom.py:668]: Sure.

**Review by jiasli — APPROVED** [review]
Core LGTM

**Review by arrownj — APPROVED** [review]

**Review by haroldrandom — APPROVED** [review]
Network is ok from my side

**Comment by qwordy** [comment]
@v-Ajnava Can you review this PR? You are code owner.

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py
Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py:3985]: I don't think this test is related with your changes. Why do you need to skip this one?

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py:3985]: Meanwhile, I can run this test lively. Anyway, I don't think we need to skip this test.

**Review by v-Ajnava — APPROVED** [review]
LGTM

File: src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py]: I reverted the change

**Pull request merged**
